rtl/eclrtl/eclhelper_base.cpp

```diff
@@ -0,0 +1,781 @@
+
+//CThorFilterArg
+bool CThorFilterArg::canMatchAny()                                   { return true; }
+bool CThorFilterArg::isValid(const void * _left)                     { return true; }
+
+//CThorFilterGroupArg
+
+bool CThorFilterGroupArg::canMatchAny()                              { return true; }
+bool CThorFilterGroupArg::isValid(unsigned _num, const void * * _rows) { return true; }
+
+//CThorIterateArg
+
+bool CThorIterateArg::canFilter()                                    { return false; }
+
+//CThorProcessArg
+
+bool CThorProcessArg::canFilter()                                { return false; }
+
+//CThorProjectArg
+
+bool CThorProjectArg::canFilter()                                { return false; }
+
+//CThorQuantileArg
+
+unsigned CThorQuantileArg::getFlags() { return 0; }
+unsigned __int64 CThorQuantileArg::getNumDivisions() { return 2; }
+double CThorQuantileArg::getSkew() { return 0; }
+unsigned __int64 CThorQuantileArg::getScore(const void * _left) { return 1; }
+void CThorQuantileArg::getRange(bool & isAll, size32_t & tlen, void * & tgt) { isAll = true; tlen = 0; tgt = NULL; }
+
+//CThorPrefetchProjectArg
+
+bool CThorPrefetchProjectArg::canFilter() { return false; }
+bool CThorPrefetchProjectArg::canMatchAny() { return true; }
+unsigned CThorPrefetchProjectArg::getFlags() { return 0; }
+unsigned CThorPrefetchProjectArg::getLookahead() { return 0; }
+IThorChildGraph *CThorPrefetchProjectArg::queryChild() { return NULL; }
+bool CThorPrefetchProjectArg::preTransform(rtlRowBuilder & extract, const void * _left, unsigned __int64 _counter) { return true; }
+
+//CThorFilterProjectArg
+
+bool CThorFilterProjectArg::canFilter() { return false; }
+bool CThorFilterProjectArg::canMatchAny() { return true; }
+
+//CThorCountProjectArg
+
+bool CThorCountProjectArg::canFilter()                                { return false; }
+
+//CThorCombineArg
+
+bool CThorCombineArg::canFilter()                                { return false; }
+
+//CThorCombineGroupArg
+
+bool CThorCombineGroupArg::canFilter()                                { return false; }
+
+//CThorActionArg
+
+void CThorActionArg::action() {}
+
+//CThorLimitArg
+
+void CThorLimitArg::onLimitExceeded() {}
+size32_t CThorLimitArg::transformOnLimitExceeded(ARowBuilder & rowBuilder) { return 0; }
+
+//CThorCatchArg
+
+unsigned CThorCatchArg::getFlags() { return 0; }
+bool CThorCatchArg::isHandler(IException * e) { return true; }
+void CThorCatchArg::onExceptionCaught() { }
+size32_t CThorCatchArg::transformOnExceptionCaught(ARowBuilder & rowBuilder, IException * e) { return 0; }
+
+//CThorSplitArg
+
+unsigned CThorSplitArg::numBranches()                          { return 2; }
+bool CThorSplitArg::isBalanced()                               { return false; }
+
+//CThorSpillArg
+
+IOutputMetaData * CThorSpillArg::queryDiskRecordSize()             { return queryOutputMeta(); }
+int CThorSpillArg::getSequence()                               { return -3; }
```

Comment:
We could delete the extra spaces at this point - I'm not sure they are very helpful.

Reply:
Done